website/examples/UHF-calculations.ipynb

```diff
@@ -0,0 +1,394 @@
+    "S = spinor_basis.quantizeOverlapOperator().parameters()\n",
+    "\n",
+    "sq_hamiltonian = gqcpy.RSQHamiltonian.Molecular(spinor_basis, molecule)  # 'sq' for 'second-quantized'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "Number of electrons: 3 \nH  (0.57735, 0, 0)\nH  (-0.288675, 0.5, 0)\nH  (-0.288675, -0.5, 0)\n\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(molecule)"
+   ]
+  },
+  {
+   "source": [
+    "## Plain UHF SCF"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "source": [
+    "Then, we need an UHF SCF solver and an UHF SCF environment."
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "environment = gqcpy.UHFSCFEnvironment.WithCoreGuess(N_alpha, N_beta, sq_hamiltonian, S)\n",
+    "plain_solver = gqcpy.UHFSCFSolver.Plain(threshold=1.0e-04, maximum_number_of_iterations=1000)  # the system is not converging very rapidly"
+   ]
+  },
+  {
+   "source": [
+    "Using GQCP's capabilities of 'injecting' Python code inside the C++ library, we'll add an intermediary step that doesn't modify the environment, but just prints out the current Fock matrices.\n",
+    "\n",
+    "This is done by writing a `Python` function that manipulates the solver's environment (of type `EnvironmentType`), and subsequently wrapping that function into a `FunctionalStep_EnvironmentType`. The instance of type `FunctionalStep_EnvironmentType` is then subsequently inserted inside the algorithm that should be performed."
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "source": [
+    "In this example, we'll print out the alpha- and beta- Fock matrices."
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def print_fock_matrices(environment):\n",
+    "    if (environment.fock_matrices_alpha):\n",
+    "        print(\"Alpha Fock matrix:\\n{}\\n\".format(environment.fock_matrices_alpha[-1]))\n",
+    "        print(\"Beta Fock matrix:\\n{}\".format(environment.fock_matrices_beta[-1]))\n",
+    "    else:\n",
+    "        print(\"No Fock matrices in the queue.\")\n",
+    "\n",
+    "print_step = gqcpy.FunctionalStep_UHFSCFEnvironment(print_fock_matrices)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plain_solver.insert(print_step, index=0)  # insert at the front"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "error",
+     "ename": "TypeError",
+     "evalue": "'method' object is not subscriptable",
```

Comment:
This notebook generates an error. Is this a known issue, or did something go wrong here?